### PR TITLE
Cleanup native dashboard, put CourseList behind flag

### PR DESCRIPTION
**Pull request by vanwagonet**
New flag nativeDashboard, so far only covers CourseList.
Dashboard got a bunch of cleanup too, but isn't even worth exposing behind the flag just yet.

refs: none
affects: Student, Teacher
release note: none

### CODE REVIEW

File: Core/Core/Courses/GetCourses.swift [hunk 1]
```diff
@@ -58,3 +58,21 @@ public class GetCourses: CollectionUseCase {
...
+
+    public var request: GetCoursesRequest {
+        return GetCoursesRequest(enrollmentState: nil, state: [ .available, .completed ], perPage: 100)
+    }
+
+    public let scope = Scope(predicate: .all, order: [
+        NSSortDescriptor(key: #keyPath(Course.isPastEnrollment), ascending: true),
+        NSSortDescriptor(key: #keyPath(Course.name), ascending: true, naturally: true),
+        NSSortDescriptor(key: #keyPath(Course.id), ascending: true),
+    ], sectionNameKeyPath: #keyPath(Course.isPastEnrollment))
```
Comment (vanwagonet) [inline Core/Core/Courses/GetCourses.swift]: this makes the same request the RN course list does.

File: Core/Core/Dashboard/DashboardGroupCell.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    func update(_ group: Group?) {
+        cardView.accessibilityIdentifier = "DashboardGroupCell.\(group?.id ?? "")"
+        cardView.accessibilityLabel = group?.name
+        accessibilityElements = [ cardView as Any ]
+        let color = group?.color.ensureContrast(against: .named(.white))
+        groupNameLabel.text = group?.name
+        // courseNameLabel.text = group?.course?.name
+        courseNameLabel.textColor = color
+        // termLabel?.text = group?.course?.term?.name
```
Comment (vanwagonet) [inline Core/Core/Dashboard/DashboardGroupCell.swift]: looks like we hadn't yet figured out how to get course info for each group. This PR doesn't try to fix that.

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/h7reasa) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fh7reasa)
teacher: [Link](https://rebrand.ly/vn3kua8) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fvn3kua8)
parent: [Link](https://rebrand.ly/wm541vv) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fwm541vv)

File: Core/CoreTests/Extensions/UIImageViewExtensionsTests.swift [hunk 3]
```diff
@@ -158,6 +158,9 @@ class UIImageViewExtensionsTests: XCTestCase {
         ], duration: 3)!
         view.load(url: pngUrl, didCompleteWith: LoadedImage(image: image, repeatCount: 2), error: nil)
         XCTAssertNil(view.loader)
+        XCTAssertNil(view.animationImages)
+        view.url = pngUrl // requires url to match for load to complete
+        view.load(url: pngUrl, didCompleteWith: LoadedImage(image: image, repeatCount: 2), error: nil)
```
Comment (vanwagonet) [inline Core/CoreTests/Extensions/UIImageViewExtensionsTests.swift]: this tests the fix for the issue where the wrong collection view cell ends up with an image

File: rn/Teacher/ios/Teacher/AppDelegate+Routes.swift [hunk 4]
```diff
@@ -31,6 +31,12 @@ extension AppDelegate {
             )
         })
 
+        if ExperimentalFeature.nativeDashboard.isEnabled {
+            HelmManager.shared.registerNativeViewController(for: "/courses", factory: { props in
```
Comment (inst-danger) [inline rn/Teacher/ios/Teacher/AppDelegate+Routes.swift]:
```suggestion
            HelmManager.shared.registerNativeViewController(for: "/courses", factory: { _ in
```

File: Core/Core/Extensions/UIImageViewExtensions.swift [hunk 5]
```diff
@@ -72,6 +72,7 @@ extension UIImageView: ImageLoadingView {
     }
 
     public func load(url: URL, didCompleteWith loaded: LoadedImage?, error: Error?) {
+        guard self.url == url else { return }
```
Comment (vanwagonet) [inline Core/Core/Extensions/UIImageViewExtensions.swift]: this fixes the issue where the wrong collection view cell ends up with an image

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student, Teacher

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/rn/Teacher/ios/Teacher/AppDelegate+Routes.swift:35:89: Unused Closure Parameter Violation: Unused parameter "props" in a closure should be replaced with _. (unused_closure_parameter)
    Linting 'ModuleSectionHeaderView.swift' (46/47)
    Linting 'Routes.swift' (47/47)
</details>
<details><summary>:x: XCTest failed: CoreTests/PersistencyTests/testAddToQueue</summary>


    Assertion Failure: PersistencyTests.swift:61: Asynchronous wait failed: Exceeded timeout of 0.5 seconds, with unfulfilled expectations: "expectation".
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**